src/dlls/mscoree/mscoree.cpp

```diff
@@ -1196,7 +1210,7 @@ STDAPI LoadLibraryShimInternal(LPCWSTR szDllName, LPCWSTR szVersion, LPVOID pvRe
 #endif // CROSSGEN_COMPILE
 
 static DWORD g_dwSystemDirectory = 0;
-static WCHAR g_pSystemDirectory[MAX_LONGPATH + 1];
+static PathString g_pSystemDirectoryPathString;
```

Comment:
This code is trying to be lock-free thread safe (see comment `use local buffer for thread safety` below). Assignment of PathString is not thread safe. You should change g_pSystemDirectory to be WCHAR *, allocate buffer of the right size using malloc, copy the result into it, and assign it to g_pSystemDirectory (pointer assignment is atomic).